session.go

```diff
@@ -273,6 +256,11 @@ func (s *Session) ProbeIP() string {
 	return ip
 }
 
+// ProxyURL returns the Proxy URL, or nil if not set
+func (s *Session) ProxyURL() *url.URL {
+	return s.proxyURL
+}
+
```

Comment:
Rather than telling experiments whether there is an explicit proxy, now we tell them what the proxy is. We cannot use a proxy function, now, because we also have non HTTP consumers, e.g. ndt7.